main.py

```diff
@@ -85,8 +94,145 @@ def recording_file(rec_file):
     run = subparsers.add_parser("run", description="Run the model on the given data")
     run.add_argument("saved_model", nargs=1, type=existing_model)
     run.add_argument("--output-file", "-o", nargs="?", default=None, type=str)
-    run.add_argument("recordings", nargs="+", type=recording_file)
+    run.add_argument("recording", nargs="+", type=recording_file)
 
     return parser.parse_args()
 
-# process_audio_directory
+
+def get_data_from_dir(data_dir, preprocess_method):
+    """
+    Loads the data from a given data directory, giving back both inputs and labels
+    :param data_dir: The directory to load data from
+    :param preprocess_method: The method to use for preprocessing (mfcc | fft)
+    :return: The inputs and labels from the data directory
+    """
+    inputs = None
+    labels = None
+
+    accent_class_folders = [folder for folder in os.listdir(data_dir)
+                            if os.path.isdir(os.path.join(data_dir, folder))]
+    for folder in accent_class_folders:
+        data_file = os.path.join(data_dir, folder, f"{folder}-{preprocess_method}.npy")
+        class_data = Io.read_audio_data(data_file)
+        class_labels = np.full((class_data.shape[0]), model.accent_classes.index(folder))
+
+        if inputs and labels:
+            inputs = np.concatenate(inputs, class_data)
+            labels = np.concatenate(labels, class_labels)
+        else:
+            inputs = class_data
+            labels = class_labels
+
+    return inputs, labels
```

Comment:
Which part of this code takes into account reading train data vs test data? And is there a format you'd like the files to be named with to read in train vs test?

Reply:
It's generalizable. The point is that you have a file structure such that it goes
train/
  raw/
    chinese/
    british/
    ...
  processed/
    chinese/
    british/
    ...
test/
  < same thing here>